src/dmd/e2ir.d

```diff
@@ -1096,8 +1096,11 @@ extern (C++) class ToElemVisitor : Visitor
             }
             else
             {
+                assert(!(global.params.ehnogc && ne.thrownew),
+                    "This should have been rewritten to `_d_newThrowable` in the semantic phase.");
```

Comment:
This assert will trigger for COM / C++ classes which are not rewritten in `expressionsem.d`?

Reply:
You're right. I changed this `assert` to the same condition as in expressionsem.d: https://github.com/dlang/dmd/blob/bf6d206751c4a5bcf20ab4c19e4a7ff283a318dc/src/dmd/expressionsem.d#L3727-L3728.

Reply:
It doesn't seem that the `!isCOMclass() && !cd.isCPPclass()` condition did not make it.